doc/devel/contributing.rst

```diff
@@ -0,0 +1,422 @@
+
+
+Retrieving and installing the latest version of the code
+========================================================
+
+
+We use `Git <https://git-scm.com/>`_ for version control and
+`GitHub <https://github.com/>`_ for hosting our main repository.
+
+You can check out the latest sources with the command::
+
+    git clone git@github.com:matplotlib/matplotlib.git
+
+
+After obtaining a local copy of the matpotlib source code (:ref:`set-up-fork`),
+navigate to the matplotlib directory and run the following in the shell:
+
+::
+    
+    python setup.py develop
+
+or::
+  
+   pip install -v -e .
+
+This installs matplotlib for development (i.e., builds everything and places the
+symbolic links back to the source code). You can then run the tests your work
+environment is set up properly::
+
+  python tests.py
+
+
+You may want to consider setting up a `virtual environment
+<http://docs.python-guide.org/en/latest/dev/virtualenvs/>`_ or a `conda
+environment <http://conda.pydata.org/docs/using/envs.html>`_
+
+
+.. note::
+  
+    If you decide to do this, you will have to rerun::
+   
+      python setup.py build
+
+    every time the source code of a compiled extension is changed (for
+    instance when switching branches or pulling changes from upstream).
+
+.. _nose: http://nose.readthedocs.org/en/latest/
+.. _pep8: http://pep8.readthedocs.org/en/latest/
+
+.. note::
+    
+  **Additional dependencies for testing**: nose_ (version 1.0 or later), `mock
+  <http://www.voidspace.org.uk/python/mock/>`_ (if python < 3.3), `Ghostscript
+  <http://pages.cs.wisc.edu/~ghost/>`_, `Inkscape <http://inkscape.org>`_
+
+.. note:: To make sure the tests run locally:
+
+      * Copy setup.cfg.template to setup.cfg
+      * Edit setup.cfg to set `test` to True, and `local_freetype` to True
+      * If you have build matplotlib previously, remove the build folder.
+      * Execute the build command.
+
+When working on bleeding edge packages, setting up a
+`virtual environment
+<http://docs.python-guide.org/en/latest/dev/virtualenvs/>`_ or a `conda
+environment <http://conda.pydata.org/docs/using/envs.html>`_ is recommended.
+
+.. seealso::
+  
+  * :ref:`testing`
+
+
+Contributing code
+=================
+
+How to contribute
+-----------------
+
+The preferred way to contribute to matplotlib is to fork the `main
+repository <https://github.com/matplotlib/matplotlib/>`__ on GitHub,
+then submit a "pull request" (PR):
+
+ 1. `Create an account <https://github.com/join>`_ on
+    GitHub if you do not already have one.
+
+ 2. Fork the `project repository
+    <https://github.com/matplotlib/matplotlib>`__: click on the 'Fork' button
+    near the top of the page. This creates a copy of the code under your
+    account on the GitHub server.
+
+ 3. Clone this copy to your local disk::
+
+        $ git clone git@github.com:YourLogin/matplotlib.git
+
+ 4. Create a branch to hold your changes::
+
+        $ git checkout -b my-feature
```

Comment:
I would say "Create a branch, normally from master..." and then append "master" to the checkout command.  I realize it is not necessary for someone following exactly these instructions, but I think it clarifies what is being done.  It gives the new user a hint as to what the line below referring to "master" might mean.